common/Environments/Models/ComputeSystem.cs

```diff
@@ -31,6 +34,48 @@ public class ComputeSystem
 
     public string DisplayName { get; private set; } = string.Empty;
 
+    [DllImport("user32.dll")]
+    [return: MarshalAs(UnmanagedType.Bool)]
+    private static extern bool AllowSetForegroundWindow(int dwProcessId);
+
+    // We need to give DevHomeAzureExtension the ability to SetForeground on the processes it creates. In some cases
+    // these processes need to show UI, in some cases they call APIs that only succeed if they are called from a
+    // foreground process. The proper way to do this is to call CoAllowSetForegroundWindow on the COM interface
+    // that we are about to use, which should automatically do all of this for us.  Unfortunately the proxy classes
+    // that we use in this file do not come with the necessary support (implementing IForegroundTransfer) by
+    // default, so we have to call AllowSetForegroundWindow ourselves.
+    // NOTE: There is currently work being done to call AllowSetForegroundWindow automatically within
+    // the DevHome SDK so that this method will no longer be necessary.  Unfortunately it won't be ready
+    // in time for Build 2024, so we need to use this method. This method should be removed once the above work
+    // is done.
+    private static bool SetAllowForegroundOnDevhomeAzureExtension()

```

Comment:
I thought about doing something similar but needing to find the process Id didn't seem manageable when there can be multiple processes with the same name. Here I doubt there would be another process with the same name unless the user has the Dev and canary versions installed

Reply:
Yep, and I went ahead and noted in the comment that multiple DevHomeAzureExtension processes running at the same time is not supported.